lib/bango/views/base.py

```diff
@@ -0,0 +1,46 @@
+from rest_framework.response import Response
+
+from lib.bango.client import format_client_error, get_client
+from lib.bango.errors import BangoFormError, BangoImmediateError, ProcessError
+from solitude.base import format_form_errors
+
+
+class BangoResource(object):
+
+    def client(self, method, data, raise_on=None, client=None):
+        """
+        Client to call the bango client and process errors in a way that
+        is relevant to the form. If you pass in a list of errors, these will
+        be treated as errors the callee is going to deal with and will not
+        be returning ImmediateHttpResponses. Instead the callee will have to
+        cope with these BangoFormErrors as appropriate.
+
+        You can optionally pass in a client to override the default.
+        """
+        raise_on = raise_on or []
+        try:
+            return getattr(client or get_client(), method)(data)
+        except BangoFormError, exc:
+            if exc.id in raise_on:
+                raise
+
+            res = self.client_errors(exc)
+            raise BangoImmediateError(format_form_errors(res))
```

Comment:
I suppose this will get caught so maybe it doesn't matter but the traceback should probably be preserved like ``raise exc, None, tb``. This might help in tracing to the original function, especially for weird proxy calls.